CMakeLists.txt

```diff
@@ -336,6 +336,19 @@ else()
   message(STATUS "CIRCT Python bindings are disabled.")
 endif()
 
+#-------------------------------------------------------------------------------
+# Tcl bindings
+#-------------------------------------------------------------------------------
+option(CIRCT_BINDINGS_TCL_ENABLED "Enables CIRCT Tcl bindings." OFF)
+
+if(CIRCT_BINDINGS_TCL_ENABLED)
+ message(STATUS "CIRCT Tcl bindings are enabled")
+ find_package(TCL 8.6 REQUIRED)
+ find_package(TCLStub 8.6 REQUIRED)
+ message(STATUS "Found Tcl and Tclstub")
```

Comment:
Tcl should be available in the Ubuntu repos, so that should just be editing the docker file somehow (I'm not familiar with that). As for the the former suggestion, I'm not sure how to get the right paths since they are system dependent.

Reply:
Re: the Docker image, the README has some guidance: https://github.com/circt/images#updating-images. Basically, you'd add the required packages to the Dockerfile in the place where we use `apt` to install dependencies. If you feel comfortable making a PR in the circt/images repo for that, go for it, otherwise someone else might be able to help out.

Re: The former suggestion, I was wondering if the `find_package(TCL ...)` call exported any variables to CMake that included the paths. For example, in Python, you get a variable called `Python3_INCLUDE_DIRS`. Basically, I was suggesting to log something like this if possible: https://github.com/llvm/circt/blob/d476344380dba7528b7389afff52068727f2f3d2/CMakeLists.txt#L325

Reply:
There are a couple interesting vars, if we want to copy the python code closely:
```cmake
message(STATUS "Found TCL include path: ${TCL_INCLUDE_PATH}")
message(STATUS "Found TCL library: ${TCL_LIBRARY}")
message(STATUS "Found TCL executable: ${TCL_TCLSH}")
```
Full documentation here: https://cmake.org/cmake/help/latest/module/FindTCL.html

Reply:
I will attempt to figure out the docker image issue. I've updated the CMAKE debug messages for Tcl, so I'll mark this as resolved and merge this PR when the checks are done.